src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -2982,12 +2982,11 @@ def create_function(cmd, resource_group_name, name, storage_account, plan=None,
     if consumption_plan_location is None and not is_plan_elastic_premium(cmd, plan_info):
         site_config.always_on = True
 
-    # If plan is elastic premium or windows consumption, we need these app settings
-    is_windows_consumption = consumption_plan_location is not None and not is_linux
-    if is_plan_elastic_premium(cmd, plan_info) or is_windows_consumption:
+    # If plan is elastic premium or consumption, we need these app settings
+    if is_plan_elastic_premium(cmd, plan_info) or consumption_plan_location is not None:
         site_config.app_settings.append(NameValuePair(name='WEBSITE_CONTENTAZUREFILECONNECTIONSTRING',
                                                       value=con_string))
-        site_config.app_settings.append(NameValuePair(name='WEBSITE_CONTENTSHARE', value=name.lower()))
+        site_config.app_settings.append(NameValuePair(name='WEBSITE_CONTENTSHARE', value=_get_content_share_name(name)))
```

Comment:
Curious can this name be an optional property exposed to users so they can offer a name & we can validate it & use it or else generate the random name.

Reply:
Yeah so this change brings content share name in line with what every other tool does (portal doesn't have a field for this, powershell client, etc...). For some reason the CLI wasn't adding that short random string which could potentially cause issues. Talking to Bala, it seems like the sentiment is that if someone wanted to get that specific it's something they could do with ARM templates. It's not that we can't/shouldn't let the customer specify a name, just that I think it would require a larger conversation across tooling